pilot/pkg/model/config.go

```diff
@@ -1016,6 +1016,19 @@ func (store *istioConfigStore) AuthenticationPolicyByDestination(hostname Hostna
 					}
 				}
 
+				// If targetSelector.subset is set in the policy, return the policy only if current service instance belongs
+				// to the subset (by comparing the serviceinstance.labels and destinationRule.subset.labels).
+				if dest.Subset != "" && instanceLabels != nil {
+					destRuleSubsetLabels := store.SubsetToLabels(dest.Subset, hostname)
+					if destRuleSubsetLabels == nil {
+						continue
+					}
+
+					if !destRuleSubsetLabels[0].SubsetOf(instanceLabels) {
+						continue
+					}
```

Comment:
This is pretty expensive isn’t it? And is there a use case for subset level auth setting?

Reply:
Do you mean store.SubsetToLabels is expensive ? My understanding is it gets destinationRule CRD so should be OK.
Security policy dark launch(see PR's referred issue) is the use case, which I'm working on a design doc. Btw this PR is prototype and I'll hold it after 0.8 release/design review.